ros_cross_compile/ros_cross_compile.py

```diff
@@ -136,13 +153,31 @@ def cross_compile_pipeline(
     args: argparse.Namespace,
 ):
     platform = Platform(args.arch, args.os, args.rosdistro, args.sysroot_base_image)
+    dc = data_collector()
 
+    # define paths for reading/writing
     ros_workspace_dir = Path(args.ros_workspace)
+    metrics_dir = Path(str(ros_workspace_dir) + '/cc_internals/metrics_data/')
     skip_rosdep_keys = args.skip_rosdep_keys
     custom_data_dir = _path_if(args.custom_data_dir)
     custom_rosdep_script = _path_if(args.custom_rosdep_script)
     custom_setup_script = _path_if(args.custom_setup_script)
 
+    # set up class wrappers for functions
+    gather_rosdeps_wrapper = pipeline_stage("gather_rosdeps", gather_rosdeps)
+    rosdep_script_wrapper = pipeline_stage("assert_install_rosdep_script_exists",
+                                           assert_install_rosdep_script_exists)
+    create_workspace_sysroot_wrapper = pipeline_stage("create_workspace_sysroot_image",
+                                                      create_workspace_sysroot_image)
+    run_docker_build_wrapper = pipeline_stage("run_emulated_docker_build",
+                                              run_emulated_docker_build)
+
+
+    # start timing of entire pipeline, and finalize metrics collection setup
+    metrics_dir.mkdir(parents=True, exist_ok=True)
```

Comment:
Management of this directory should happen in the data collector object, not out here in the global context